### PR TITLE
Add accept and reject symfony events for share

### PR DESCRIPTION
**Pull request by sharidas**
Add symfony events for accept and decline of shares.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

## Description

Add symfony events for 
- accept of share
- decline of share

## Related Issue

https://github.com/owncloud/core/issues/31550

## Motivation and Context

Add symfony events for 
- accept of share
- decline of share

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: apps/files_sharing/lib/API/Share20OCS.php [hunk 1]
```diff
@@ -839,17 +845,35 @@ public function updateShare($id) {
 	 * @return \OC\OCS\Result
 	 */
 	public function acceptShare($id) {
-		return $this->updateShareState($id, \OCP\Share::STATE_ACCEPTED);
+		$result = $this->updateShareState($id, \OCP\Share::STATE_ACCEPTED);
+		if (!empty($result->getData())) {
+			$this->eventDispatcher->dispatch('share.accepted',
+				new GenericEvent(null,
+					['share' => $this->getShareById($id, $this->currentUser->getUID())]));
```
Comment (PVince81) [inline apps/files_sharing/lib/API/Share20OCS.php]: we might be able to reuse the existing instance of IShare as we already have it in the `updateShareState` method in some form, maybe move the event there instead ?

**Review by PVince81 — COMMENTED** [review]
see comment

File: apps/files_sharing/lib/API/Share20OCS.php [hunk 2]
```diff
@@ -850,7 +856,19 @@ public function declineShare($id) {
...
 	}
 
+	/**
+	 * @param $id
+	 * @param $state
+	 * @return \OC\OCS\Result
+	 */
 	private function updateShareState($id, $state) {
+		$eventName = '';
+		if ($state === \OCP\Share::STATE_ACCEPTED) {
```
[the remark this answers is not in the document]
Comment (sharidas) [reply apps/files_sharing/lib/API/Share20OCS.php]: Introduced before/after pattern here.

**Review by PVince81 — COMMENTED** [review]
see comments

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by PVince81** [comment]
Please backport

**Comment by sharidas** [comment]
Backport PR: https://github.com/owncloud/core/pull/31702

**Pull request merged**